### PR TITLE
CLI-47 support multiple files & dir analysis with secrets

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,23 +259,33 @@ Analyze code for security issues
...
 
 **Options:**
 
-| Option    | Type    | Required | Description                                | Default |
-| --------- | ------- | -------- | ------------------------------------------ | ------- |
-| `--file`  | string  | No       | File path to scan for secrets              | -       |
-| `--stdin` | boolean | No       | Read from standard input instead of a file | -       |
+| Option    | Type    | Required | Description                                 | Default |
+| --------- | ------- | -------- | ------------------------------------------- | ------- |
+| `--stdin` | boolean | No*      | Read from standard input instead of paths  | -       |
```
Comment (jonas-wielage-sonarsource) [inline README.md]: I think this is generated, so not sure if we can influence this, but the table is off one space :)

File: src/cli/commands/integrate/claude/hook-templates.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ if [[ -z "$file_path" ]] || [[ ! -f "$file_path" ]]; then
 fi
 
 # Scan file for secrets
-sonar analyze --file "$file_path" > /dev/null 2>&1
+sonar analyze secrets "$file_path" > /dev/null 2>&1
```
Comment (jonas-wielage-sonarsource) [inline src/cli/commands/integrate/claude/hook-templates.ts]: Nice that this is fixed :)

File: tests/unit/secret-scan.test.ts
Comment (jonas-wielage-sonarsource) [inline tests/unit/secret-scan.test.ts:1]: We could also add some tests with empty paths array or 2 files inside the array

File: tests/unit/secret.test.ts
Comment (jonas-wielage-sonarsource) [inline tests/unit/secret.test.ts:1]: We could also add some tests with empty paths array or 2 files inside the array

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, thanks for the quick improvement here 💪 
Please have a look at my comments to see if it's worth adding before merging

File: src/cli/command-tree.ts [hunk 3]
```diff
@@ -162,10 +162,14 @@ const analyze = COMMAND_TREE.command('analyze').description('Analyze code for se
...
-  .description('Scan a file or stdin for hardcoded secrets')
-  .option('--file <file>', 'File path to scan for secrets')
-  .option('--stdin', 'Read from standard input instead of a file')
-  .action((options: AnalyzeSecretsOptions) => runCommand(() => analyzeSecrets(options)));
+  .description('Scan files or stdin for hardcoded secrets')
+  .argument('[paths...]', 'File or directory paths to scan for secrets')
+  .option('--stdin', 'Read from standard input instead of paths')
+  .action((paths: string[], options: AnalyzeSecretsOptions) =>
+    runCommand(() =>
+      analyzeSecrets({ paths: Array.isArray(paths) ? paths : [], stdin: options.stdin }),
```
Comment (kirill-knize-sonarsource) [inline src/cli/command-tree.ts:172]: As discussed we will need to think about coherent UX for all the analysis we will have. What happens if people call just parent command:
`sonar analyze file1.txt file2.txt`
What happens to analyzing commands who doesn't support multi-file analysis?

File: build-scripts/examples.ts [hunk 4]
```diff
@@ -99,7 +99,7 @@ export const EXAMPLES: Record<string, Example[]> = {
   ],
   'sonar analyze secrets': [
     {
-      command: 'sonar analyze secrets --file src/config.ts',
+      command: 'sonar analyze secrets src/config.ts',
```
Comment (kirill-knize-sonarsource) [inline build-scripts/examples.ts:102]: There's no example that shows we can call with many files.

**Review by kirill-knize-sonarsource — APPROVED** [review]
LGTM
